enhancements/audit-command.md

```diff
@@ -0,0 +1,594 @@
+```
+
+Following how we would set the values for the above example. 
+
+| Column | Description | 
+| ------ | ----- | 
+| Builder | `SDK`|
+| Language | `golang` |
+| SDK Version | `v0.17.0` |
+| Golang Version | `1.13` |
+
+**NOTE** If the `go.mod` was found but not the SDK dependency we will set the `Language` as `golang`. Also, we will always check the `go.mod` file for Golang projects to know the Golang version used.  
+
+**Gathering the info from build/Dockerfile**
+
+You will only need to check the `build/Dockerfile` when the project is not writen in Go. In this way, when the `go.mod` file cannot to be found then, it might to be an operator project using Helm or Ansible which was built with an SDK version < 1.0. In this case, we will be looking for `quay.io/operator-framework/ansible-operator` or `quay.io/operator-framework/helm-operator` images. Let's check an Ansible Operator. See the [Dockerfile](https://github.com/operator-framework/operator-sdk-samples/blob/v0.17.0/ansible/memcached-operator/build/Dockerfile): 
+
+```
+FROM quay.io/operator-framework/ansible-operator:v0.17.0
+```
+
+| Column | Description | 
+| ------ | ----- | 
+| Builder | `SDK`|
+| Language | `ansible` |
+| SDK Version | `v0.17.0` |
+
+Now, see a [Dockerfile](https://github.com/operator-framework/operator-sdk-samples/blob/v0.17.0/helm/memcached-operator/build/Dockerfile) for a Helm operator project built with SDK version < 1.0. Then, note that the same fields will be obtained:
+
+| Column | Description | 
+| ------ | ----- | 
+| Builder | `SDK`|
+| Language | `helm` |
+| SDK Version | `v0.17.0` |
+
+### Auditing just the latest channel heads 
+
+This option allows the user audit only the latest bundle versions (default channel heads) for each package. It can be enabled by setting `--head-only` boolean flag.
+
+By using this flag, the audit command will filter the operator bundles in the first step to download and extract just this ones. 
+
+To know the operator bundles which are the latest head we will get only the entries which has values in the `operatorbundle` table for both `csv` and `bundle` column. 
+
+### Disabling the data to perform only the checks
+
+By using the option `--data=false`, the audit command will only be gathering the operator bundle names and run the checks. All other columns described above will be skipped.
+
+This option is useful for who would like to add the audit in an automated CI job for example to check the catalog.  
+ 
+### Disabling the warning(s)
+
+By using the option `--warning=false`, the audit command will only raise the errors  found.
+
+This option is useful for who would like to add the audit in an automated CI job for example to check the catalog combined with the flag `--data=false`.  
+ 
+### Risks and Mitigations
+
+A risk is the command take to long to be executed. In order to mitigate it, the flags `--wait-time` and customize the checks that should be done as the data that could be used was added: 
```

Comment:
`--wait-time` should be configured on a per-bundle or per-test basis, since the caller would have to calculate `wait time * bundles in catalog * number of tests` to get a reasonable estimate for an overall wait time.

Reply:
The idea is we have a default timeout set up for the command and then, it can be changed by the flag for who is using it and has a better idea over what would be a reasonable time or not for the case checked. 

I do not think that we should try to guess a default timeout based on the bundles and etc. It can change a lot since the user might be running the full report or not, might be using the checks or not and etc. So, to keep the simplicity IMO the best approach is to define a default time and let people configure it as they wish.

Reply:
Addressed in the EP as well. So, I am closing, however,  please feel free to re-open if you wish.

Reply:
I disagree that having an overall timeout is useful, unless you set it to 10 hours or something ridiculous.

Reply:
For the alpha version, I do not think that we need or should to address this kind of complexity. I'd prefer the timeout as 2 hours by default instead. We are allowing users to change it as please them which shows enough.  I will try to clarify in the open questions.